janitor/finance.py

```diff
@@ -204,3 +487,291 @@ def convert_currency(
         df[column_name] = df[column_name] * rate
 
     return df
+
+
+@lru_cache(maxsize=32)
+def _inflate_currency(
+    country: str = None, currency_year: int = None, to_year: int = None
+):
+    """
+    Currency inflation for Pandas DataFrame column.
+    Helper function for `inflate_currency` method.
+    The API used is the World Bank Indicator API:
+    https://datahelpdesk.worldbank.org/knowledgebase/articles/889392-about-the-indicators-api-documentation
+    """
+
+    # Check all inputs are correct data type
+    check("country", country, [str])
+    check("currency_year", currency_year, [int])
+    check("to_year", to_year, [int])
+
+    # Get WB country abbreviation
+    _check_wb_country(country)
+    if country in wb_country_dict.keys():
+        country = wb_country_dict[country]
+    else:
+        # `country` is already a correct abbreviation; do nothing
+        pass
+
+    _check_wb_years(currency_year)
+    _check_wb_years(to_year)
+
+    url = (
+        "https://api.worldbank.org/v2/country/"
+        + country
+        + "/indicator/FP.CPI.TOTL.ZG?date="
+        + str(min(currency_year, to_year))
+        + ":"
+        + str(max(currency_year, to_year))
+        + "&format=json"
+    )
+
+    result = requests.get(url)
+
+    if result.status_code != 200:
+        raise ConnectionError(
+            "WB Indicator API failed to receive a 200 "
+            "response from the server. "
+            "Please try again later."
+        )
+
+    # The API returns a list of two items;
+    # the second item in the list is what we want
+    inflation_dict = json.loads(result.text)[1]
+
+    # Error checking
+    if inflation_dict is None:
+        raise ValueError(
+            "The WB Indicator API returned nothing. "
+            "This likely means the currency_year and "
+            "to_year are outside of the year range for "
+            "which the WB has inflation data for the "
+            "specified country."
+        )
+
+    # Create new dict with only the year and inflation values
+    inflation_dict_ready = {
+        int(inflation_dict[i]["date"]): float(inflation_dict[i]["value"])
+        for i in range(len(inflation_dict))
+        if inflation_dict[i]["value"] is not None
+    }
+
+    # Error catching
+    if currency_year not in inflation_dict_ready.keys():
+        raise ValueError(
+            f"The WB Indicator API does not have inflation "
+            f"data for {currency_year} for {country}."
+        )
+    if to_year not in inflation_dict_ready.keys():
+        raise ValueError(
+            f"The WB Indicator API does not have inflation "
+            f"data for {to_year} for {country}."
+        )
+
+    inflator = (
+        inflation_dict_ready[to_year] / inflation_dict_ready[currency_year]
+    )
+    return inflator
+
+
+@pf.register_dataframe_method
+@deprecated_alias(colname="column_name")
```

Comment:
`inflate_currency()` shouldn't need the `deprecated_alias` decorator since it is a new method. You only need it if it once accepted `colname` in the past

Reply:
Thanks @hectormz.  I wasn't entirely sure what that decorator was for, but I saw it attached to many existing methods.  I've removed it from this new inflate_currency() method.